food-remedy-co/frc-api-cyber-security-best-practices.md

```diff
@@ -0,0 +1,451 @@
+- Example: An attacker finds a commit in a public repository with a plaintext AWS access key.
+
+### 2. Code Sharing Platforms:
+
+- Developers sometimes share code snippets containing credentials on platforms like Pastebin or Gist.
+- Attackers use Google Dorks (advanced search queries) to find publicly shared credentials.
+- Example: A developer inadvertently shares a Python script with hardcoded API keys on Pastebin.
+
+### OSINT Tools
+
+- **TruffleHog:** Searches for high entropy strings and keywords in Git repositories.
+- **GitRob:** Scans GitHub repositories for exposed sensitive information.
+- **Google Dorks:** Custom search queries to find specific information using Google's search engine.
+
+### Scraping Tools
+
+- **Web Scrapers:** Tools like Scrapy or Beautiful Soup automate data extraction from websites.
+- **Malicious Bots:** Attackers deploy malicious bots to scrape and collect data from websites.
+
+### Advanced Google Search
+
+Advanced Google search techniques can help you identify instances of security hardcoded credentials online.
+
+- **Identify Keywords:** Start by identifying relevant keywords that are commonly used in hardcoded credentials. These could include terms like "password," "secret," "API key," "access token," etc.
+- **Use Quotation Marks:** Wrap your search terms in quotation marks to search for exact phrases. For example, search for "password=12345" or "API key=abcdef".
+- **Exclude Common Pages:** To avoid common pages like documentation and tutorials, you can exclude specific domains from your search. For example, -site:github.com will exclude results from GitHub.
+- **File Type Search:** Search for specific file types that often contain configuration files with hardcoded credentials. Use the filetype: operator, such as filetype:xml password.
+- **Inurl Operator:** Use the inurl: operator to search for specific terms in the URL of a page. For example, inurl:config secret.
+- **Wildcard Operator:** Use the * wildcard to capture variations of a term. For instance, "API key=*" will search for different API key values.
+- **Intitle Operator:** Use the intitle: operator to find pages with specific terms in their title. For example, intitle:"login credentials".
+- **Boolean Operators:** Combine operators like AND, OR, and NOT to refine your search. For example, password OR "API key" NOT tutorial.
+- **Search Specific Domains:** Target specific domains that might be relevant to your search, like specific websites or forums where developers discuss code.
+
+**Example:** "password=12345" OR "API key=abcdef" -site:github.com -site:stackoverflow.com filetype:xml
+
+Additionally, you might consider using dedicated tools and platforms designed for searching for leaked or exposed credentials on the internet, as they can provide more focused and comprehensive results. Examples of such tools include Shodan, Censys, and Have I Been Pwned.
+
+### Points to Keep in Mind
```

Comment:
Formatting issue, consider a tab space for the sub-point and clear the extra line in between the Main point and sub-points.